packages/playwright-test/src/worker/testInfo.ts

```diff
@@ -109,7 +109,7 @@ export class TestInfoImpl implements TestInfo {
 
   _deadlineForMatcher(timeout: number): { deadline: number, timeoutMessage: string } {
     const startTime = monotonicTime();
-    const matcherDeadline = timeout ? startTime + timeout : MaxTime;
+    const matcherDeadline = timeout ? startTime + timeout : MaxSetIntervalValue;
```

Comment:
Deadline can't be capped with MaxSetIntervalValue if monotonicTime can be bigger than MaxSetIntervalValue.